internal/controller/rolebindingtemplate_controller.go

```diff
@@ -0,0 +1,322 @@
+	}
+
+	return namespaces, nil
+}
+
+func (r *RoleBindingTemplateReconciler) getNamespacesFromOrganizations(ctx context.Context, organizations *security.OrganizationList) ([]string, error) {
+	namespaces := []string{}
+	for _, o := range organizations.Items {
+		if o.Status.Namespace != "" {
+			// get the org namespace
+			namespaces = append(namespaces, o.Status.Namespace)
+		}
+
+		// get the cluster namespaces that belong to the org namespace
+		labelSelector, err := labels.Parse(fmt.Sprintf("%s=%s,%s", label.Organization, o.Name, label.Cluster))
+		if err != nil {
+			return nil, err
+		}
+
+		clusterNamespaces := &corev1.NamespaceList{}
+		if err := r.List(ctx, clusterNamespaces, &client.ListOptions{LabelSelector: labelSelector}); err != nil {
+			return nil, err
+		}
+
+		for _, cns := range clusterNamespaces.Items {
+			namespaces = append(namespaces, cns.Name)
+		}
+	}
+	return namespaces, nil
+}
+
+func getRoleBindingNameFromTemplate(template *v1alpha1.RoleBindingTemplate) string {
+	roleBindingName := template.Spec.Template.Name
+	if roleBindingName == "" {
+		roleBindingName = template.Name
+	}
+	return roleBindingName
+}
+
+func getRoleBindingFromTemplate(template *v1alpha1.RoleBindingTemplate, namespace string) *rbacv1.RoleBinding {
+	roleBinding := rbacv1.RoleBinding{
+		ObjectMeta: template.Spec.Template.ObjectMeta,
+		RoleRef:    template.Spec.Template.RoleRef,
+	}
+	roleBinding.Name = getRoleBindingNameFromTemplate(template)
+	roleBinding.Namespace = namespace
+
+	if roleBinding.Labels == nil {
+		roleBinding.Labels = map[string]string{}
+	}
+	roleBinding.Labels[label.ManagedBy] = project.Name()
+
+	if roleBinding.Annotations == nil {
+		roleBinding.Annotations = map[string]string{}
+	}
+	if roleBinding.Annotations[annotation.Notes] == "" {
+		roleBinding.Annotations[annotation.Notes] = fmt.Sprintf("Generated based on RoleBindingTemplate %s", template.Name)
+	}
+
+	if roleBinding.RoleRef.APIGroup == "" {
+		roleBinding.RoleRef.APIGroup = "rbac.authorization.k8s.io"
+	}
+
+	subjects := []rbacv1.Subject{}
+	{
+		for _, subject := range template.Spec.Template.Subjects {
+			if subject.Kind == rbacv1.ServiceAccountKind && subject.Namespace == "" {
+				subject.Namespace = namespace
+			}
+			subjects = append(subjects, subject)
+		}
+	}
+	roleBinding.Subjects = subjects
+
+	return &roleBinding
+}
+
+func cleanSubjects(roleBinding *rbacv1.RoleBinding, namespace string) *rbacv1.RoleBinding {
+	// if the rolebinding is in a protected namespace, subjects can only be serviceAccounts in flux namespace or the same namespace
+	if !pkgkey.IsProtectedNamespace(namespace) {
+		return roleBinding
+	}
+	var validSubjects []rbacv1.Subject
+	for _, subject := range roleBinding.Subjects {
+		if subject.Kind == rbacv1.ServiceAccountKind && (subject.Namespace == pkgkey.FluxNamespaceName || subject.Namespace == namespace) {
+			validSubjects = append(validSubjects, subject)
+		}
+	}
+	roleBinding.Subjects = validSubjects
+	return roleBinding
```

Comment:
I realize this decision pre-dates you being here, but I'm wondering about it.

1. just checking -- as far as you know, this behavior matches the operatorkit implementation?
2. if it were my choice, I would not silently "clean" a bad RBT, but rather reject it. Do you have any suspicion as to why it was done this way originally?
3. WDYT about putting this in the key file versus keeping it here? I'm ambivalent.